activiti-cloud-query-service/activiti-cloud-starter-query/src/main/resources/config/query/liquibase/changelog/07-alter.oracle.schema.7.1.0-m11.sql

```diff
@@ -0,0 +1,13 @@
+create table application
+(
+    id                         varchar(255) not null,
+    name                       varchar(255) not null,
+    version                    varchar(255),
+    primary key (id)
+);
+
+INSERT INTO application
+SELECT DISTINCT CONCAT(app_name, app_version), app_name, app_version
+FROM process_instance;
```

Comment:
I think it's better to use `process_definition` table as it also has `app_name` and `app_version`, but has fewer entries than `process_instance`. In addition, I'm wondering if this step is actually needed: with the current implementation, runtime bundle will send those events every time it starts, so basically this script will only be useful if we migrate query-service without migrating runtime bundle, what doesn't seem to be a real use case. What do you think?

Reply:
I did that thinking a possible scenario where there are process instances saved before adding the handler for the event of application deployed in  Query. Happy to remove the script if this case will never happen in a real case. Thanks

Reply:
There will be process instances already saved from previous versions, but on every start of the runtime bundle, it will send an ApplicationDeployedEvent for all deployments with the name `SpringAutoDepoyment`, even those that existed before. In the case where the runtime bundle starts before query service and sends the event before query is up and running, the message broker will keep the message and will deliver it to query when it starts listening to the events. So it should be safe to remove this... we can double-check by doing an application upgrade once it's merged.